cmd/translatesfx/translatesfx/directive.go

```diff
@@ -0,0 +1,138 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translatesfx
+
+import (
+	"fmt"
+	"strings"
+)
+
+type source int
+
+const (
+	directiveSourceUnknown source = iota
+	directiveSourceFile
+	directiveSourceEnv
+	directiveSourceZK
+	directiveSourceEtcd2
+	directiveSourceConsul
+	directiveSourceVault
+)
+
+type directive struct {
+	fromPath    string
+	defaultV    string
+	fromType    source
+	isDirective bool
+	flatten     bool
+	optional    bool
+}
+
+func parseDirective(m map[interface{}]interface{}) (out directive, err error) {
+	fromRaw, ok := m["#from"]
+	if !ok {
+		return
+	}
+	out.isDirective = true
+	from, ok := fromRaw.(string)
+	if !ok {
+		return out, fmt.Errorf("error parsing from %v", m)
+	}
+
+	out.fromType, out.fromPath, err = parseFrom(from)
+	if err != nil {
+		return
+	}
+
+	out.flatten, err = parseFlatten(m)
+	if err != nil {
+		return
+	}
+
+	out.defaultV, err = parseDefault(m)
+	if err != nil {
+		return
+	}
+
+	out.optional, err = parseOptional(m)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func parseFrom(from string) (s source, target string, err error) {
+	idx := strings.Index(from, ":")
+	if idx == -1 {
+		return directiveSourceFile, from, nil
+	}
+	s = strToSource(from[:idx])
+	if s == directiveSourceUnknown {
+		return s, target, fmt.Errorf("#from fromType type unknown: %s", from)
+	}
+	target = from[idx+1:]
+	return
```

Comment:
nit, mixing named and unnamed returns makes this more confusing to me than necessary.

Reply:
Thanks for the feedback -- fixed.